src/aes.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::marker::PhantomData;
+
+use crate::traits::{AuthenticatedCipher, Cipher, EncryptionKey, ToFromBytes};
+
+use aes::cipher::{BlockDecryptMut, BlockEncryptMut};
+use aes_gcm::{AeadInPlace, KeyInit};
+use ctr::cipher::StreamCipher;
+use digest::{crypto_common::KeyIvInit, generic_array::ArrayLength, typenum::U16};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+
+///
+/// An encryption key for a symmetric cipher consisting of `N` bytes.
+///
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Key<const N: usize> {
```

Comment:
Similarly to what we do for other schemes (i.e., for [ed25519](https://github.com/MystenLabs/fastcrypto/blob/main/src/ed25519.rs#L42)), let's rename this to `AesKey` to avoid any future readability confusion against future apis ie HMAC keys etc.